src/vs/workbench/parts/lib/node/settingsVariables.ts

```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as Types from 'vs/base/common/types';
+import { AbstractSystemVariables } from 'vs/base/common/parsers';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+
+export class SettingsVariables extends AbstractSystemVariables {
+	private configuration: any;
+	constructor(private configurationService: IConfigurationService) {
+		super();
+		this.configuration = this.configurationService.getConfiguration();
+	}
+
+	protected resolveString(value: string): string {
+		let regexp = /\$\{settings\.(.*?)\}/g;
```

Comment:
I would prefer if you just read the configuration every time you need to resolve the string. 
Not sure if doing it the current way you can end up with a stale configuration.